packages/foam-vscode/src/features/tags-tree-view/index.ts

```diff
@@ -82,6 +82,16 @@ export class TagsProvider implements vscode.TreeDataProvider<TagTreeItem> {
 
       const references: TagTreeItem[] = element.notes
         .map(({ uri }) => this.foam.workspace.get(uri))
+        .filter(note =>
+          // foreach nested tag item check if the exact tag is present as a tag on the note. If so,
+          // it belongs to a nested tag item and should not be displayed as a reference underneath the current element
+          nestedTagItems.reduce(
```

Comment:
IIUC what you are trying to achieve, could you use `nestedTagItems.find`?
Then depending on what you need, you can use `isSome` or `isNone`

Reply:
Ah, that might be a nice solution. Was working with my limited knowledge and didn't know about that. Will have a look.